src/app/organizer/Columns.tsx

```diff
@@ -0,0 +1,419 @@
+/* eslint-disable react/jsx-key, react/prop-types */
+import React from 'react';
+import { DimItem, DimPlug } from 'app/inventory/item-types';
+import { Row, UseRowSelectRowProps, UseRowSelectInstanceProps } from 'react-table';
+import BungieImage from 'app/dim-ui/BungieImage';
+import {
+  AppIcon,
+  powerIndicatorIcon,
+  lockIcon,
+  thumbsUpIcon,
+  thumbsDownIcon
+} from 'app/shell/icons';
+import styles from './ItemTable.m.scss';
+import ItemPopupTrigger from 'app/inventory/ItemPopupTrigger';
+import _ from 'lodash';
+import { getTag, getNotes, tagConfig } from 'app/inventory/dim-item-info';
+import TagIcon from 'app/inventory/TagIcon';
+import { source } from 'app/inventory/spreadsheets';
+import ElementIcon from 'app/inventory/ElementIcon';
+import { D2SeasonInfo } from 'app/inventory/d2-season-info';
+import { D2EventInfo } from 'data/d2/d2-event-info';
+import { getRating } from 'app/item-review/reducer';
+import { statWhiteList } from 'app/inventory/store/stats';
+import { compareBy } from 'app/utils/comparators';
+import { rarity } from 'app/shell/filters';
+import { faCheck } from '@fortawesome/free-solid-svg-icons';
+import RatingIcon from 'app/inventory/RatingIcon';
+import { getItemSpecialtyModSlotDisplayName } from 'app/utils/item-utils';
+import SpecialtyModSlotIcon from 'app/dim-ui/SpecialtyModSlotIcon';
+import { DestinyCollectibleState } from 'bungie-api-ts/destiny2';
+import CompareStat from 'app/compare/CompareStat';
+import { StatInfo } from 'app/compare/Compare';
+import { filterPlugs } from 'app/loadout-builder/generated-sets/utils';
+import PressTip from 'app/dim-ui/PressTip';
+import PlugTooltip from 'app/item-popup/PlugTooltip';
+import { INTRINSIC_PLUG_CATEGORY } from 'app/inventory/store/sockets';
+import { DimColumn } from './ItemTable';
+
+export const initialEnabledColumns = [
+  'selection',
+  'icon',
+  'name',
+  'dmg',
+  'power',
+  'locked',
+  'tag',
+  'wishList',
+  'rating',
+  'archetype',
+  'perks',
+  'mods',
+  'notes'
+];
+
+// TODO: drop wishlist columns if no wishlist loaded
+// TODO: d1/d2 columns
+// TODO: stat ranges
+// TODO: special stat display? recoil, bars, etc
+
+// TODO: really gotta pass these in... need to figure out data dependencies
+// https://github.com/tannerlinsley/react-table/blob/master/docs/api.md
+export function getColumns(itemInfos, ratings, wishList, defs, items): Map<string, DimColumn> {
```

Comment:
Each of these parameters needs a type.

I'm also starting to suspect that this columns function should be part of a redux `mapStateToProps` function somewhere.